sky/backends/backend_utils.py

```diff
@@ -2879,3 +2883,50 @@ def check_stale_runtime_on_remote(returncode: int, stderr: str,
                     f'not interrupted): {colorama.Style.BRIGHT}sky start -f -y '
                     f'{cluster_name}{colorama.Style.RESET_ALL}'
                     f'\n--- Details ---\n{stderr.strip()}\n')
+
+
+def wait_and_terminate_csync(cluster_name: str) -> None:
+    """Terminates all the CSYNC process running in each node.
+
+    Before terminating the CSYNC daemon, it waits until the sync process
+    launched by CSYNC is completed if there are any.
+
+    Args:
+        cluster_name: Cluster name (see `sky status`)
+    """
+    record = global_user_state.get_cluster_from_name(cluster_name)
+    assert record is not None, cluster_name
+    handle = record['handle']
+    if not isinstance(handle, backends.CloudVmRayResourceHandle):
+        return
+    try:
+        ip_list = handle.external_ips()
+    # When cluster is in INIT status, attempt to fetch IP fails raising an error
+    except exceptions.FetchIPError:
+        return
+    port_list = handle.external_ssh_ports()
+    ssh_credentials = ssh_credential_from_yaml(handle.cluster_yaml,
+                                               handle.docker_user)
```

Comment:
1. Confirmed that `CSYNC` works well when ran with [docker run time](https://skypilot.readthedocs.io/en/latest/examples/docker-containers.html#using-containers-as-runtime-environments)
2. `local_docker_backend` only supports for local file mounts, and the check to prevent cloud storage mounts is done at `local_docker_backend.py/_sync_file_mounts`.